src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -330,6 +330,10 @@ private IntermediateNode[] RewriteUsage(IntermediateNode parent, BindEntry bindE
             {
                 var valueNode = new HtmlAttributeIntermediateNode()
                 {
+                    Annotations =
+                    {
+                        [ComponentMetadata.Common.OriginalNode] = node,
```

Comment:
Just as something we should be aware of: without profiling i'd imagine this significantly increases the size of our IR tree since the original nodes can no longer be garbage collected. If we end up seeing issues in the future we may want to consider only tracking the original node's attribute name since that's all this is used for today. Food for thought.

Reply:
Sure, good plan.